src/aux-records/AIController.spec.ts

```diff
@@ -369,6 +369,86 @@ describe('AIController', () => {
             expect(chatInterface.chat).not.toBeCalled();
         });
 
+        it('should return success when allowedModels includes the model', () => {
+            const allowedFeatures = {
+                ai: {
+                    chat: {
+                        allowedModels: ['modelA', 'modelB'],
+                    },
+                },
+            };
+            const model = 'modelA';
+            const result = { choices: ['choice1', 'choice2'] };
+
+            const response = (function () {
```

Comment:
@TroyceGowdy These tests need to actually call into the AIController in order to properly test that the AIController is  functioning correctly. Currently, the test just happens to use the same code as the AIController, but there is no guarantee that it will stay that way in the future,